internal/state/errors.go

```diff
@@ -5,6 +5,7 @@ import "fmt"
 type (
 	ErrInvalidBlock error
 	ErrProxyAppConn error
+	ErrPrepareProposalValidationFailed error
```

Comment:
I was initially thinking we might need to act based on the error type returned. But, since we're "panic"ing in all cases, we don't use the error type.
So, I removed it as you were suggesting.